numpy/lib/nanfunctions.py

```diff
@@ -663,12 +663,9 @@ def nanmean(a, axis=None, dtype=None, out=None, keepdims=False):
     if out is not None and not issubclass(out.dtype.type, np.inexact):
         raise TypeError("If a is inexact, then out must be inexact")
 
-    # The warning context speeds things up.
-    with warnings.catch_warnings():
```

Comment:
Didn't understand why it would speed things up, I somewhat assumed that adding divide=ignore for _divide_by_count does the trick here as well. (same below)